### PR TITLE
63 app movie pages

### PR DESCRIPTION
**Pull request by Julien-Goletto**
Movie page is done !
Can be improved of course, but satisfied by this foundation.

### CODE REVIEW

**Comment by BenoitSafari** [comment]
You do not use PR system correctly, you should wait for merge before creating a new branch and not checkout from a branch that IS in review. If you can't wait for my review then you should force merge.
Also, if you do more PR then I can check theme quicker.

File: app/components/Input/Interaction/index.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  const SvgComponent = (type === 'bookmarked') ? BookmarkSvg : (type === 'liked') ? LikeSvg : ViewSvg;
+  return (
+    <button
+      onClick={onClick}
+      className={isClicked ? buttonStyle + 'interaction--clicked translate-y-[2px]' : buttonStyle}>
+      <SvgComponent style={isClicked ? svgStyle + 'fill-white' : svgStyle} />
+      <p className="text-sm mt-1">{counter}</p>
+    </button>
+  );
+};
```
Comment (BenoitSafari) [inline app/components/Input/Interaction/index.tsx:43]: 1 component = 1 file

File: app/components/Input/Interaction/index.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
+import type { FormEvent } from 'react';
+import React, { useRef } from 'react';
```
Comment (BenoitSafari) [inline app/components/Input/Interaction/index.tsx:2]:
```suggestion
import { useRef } from 'react';
```

File: app/components/Input/Interaction/StarRadio.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import type { FormEvent } from 'react';
+import React from 'react';
```
Comment (BenoitSafari) [inline app/components/Input/Interaction/StarRadio.tsx:2]:
```suggestion
```

File: app/components/Input/Interaction/SVG.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+
```
Comment (BenoitSafari) [inline app/components/Input/Interaction/SVG.tsx:2]:
```suggestion
```

File: app/components/Input/Form.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import React, { useState } from 'react';
```
Comment (BenoitSafari) [inline app/components/Input/Form.tsx:1]:
```suggestion
import { useState } from 'react';
```

File: app/components/Input/CheckBox.tsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        name={id}
+        {...props}
+        className="w-5 h-5 border border-transparent rounded 
+        bg-orange-primary/20 text-orange-secondary
+        focus:border-orange-primary focus:ring-4 focus:ring-offset-0 focus:ring-white/5
+        hover:border-orange-primary hover:ring-4 hover:ring-offset-0 hover:ring-white/5"
+      />
+    </div>
+  );
+};
```
Comment (BenoitSafari) [inline app/components/Input/CheckBox.tsx:40]: 1 component = 1 file

File: app/components/Input/Button.tsx [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+      break;
+    case 'rounded':
+      className += ' ' + rounded;
+      break;
+    case 'white':
+      className += ' ' + white;
+      break;
+    default:
+      className += ' ' + classic;
+  }
```
Comment (BenoitSafari) [inline app/components/Input/Button.tsx:52]: I'm not a hudge fan of this, there's probably a cleaner way. (Don't try to improve that for 2 days without eating, it's not important!!)

File: app/components/Input/Button.tsx [hunk 8]
```diff
@@ -0,0 +1,81 @@
+import React from 'react';
```
Comment (BenoitSafari) [inline app/components/Input/Button.tsx:1]:
```suggestion
```

File: app/components/Forms/InscriptionForm.tsx [hunk 9]
```diff
@@ -0,0 +1,129 @@
+import React, { useRef } from 'react';
```
Comment (BenoitSafari) [inline app/components/Forms/InscriptionForm.tsx:1]:
```suggestion
import { useRef } from 'react';
```

File: app/components/Forms/InscriptionForm.tsx [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+
+  const emailRef = useRef<HTMLInputElement>(null);
+  const usernameRef = useRef<HTMLInputElement>(null);
+  const PWRef = useRef<HTMLInputElement>(null);
+  const confirmPWRef = useRef<HTMLInputElement>(null);
+  const allInputsRef = [emailRef, usernameRef, PWRef, confirmPWRef];
+
+  const helpingTextStyle = 'px-1 text-sm font-light italic text-gray-300';
+
+  let arePWMatching = true;
```
Comment (BenoitSafari) [inline app/components/Forms/InscriptionForm.tsx:26]: Why not use a `state`?

File: app/components/Forms/ConnectionForm.tsx [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+    // Checking all inputs validation status
+    const inputValidationStatus = allInputsRef.map((inputRef) => inputRef.current?.reportValidity());
+    if (!inputValidationStatus.includes(false)) {
+      const data = {
+        password: passwordRef.current!.value,
+        pseudo: identifierRef.current!.value,
+      };
+      tryCatchWrapper(submitSuccess)('POST', '/login', data);
+    }
+  };
```
Comment (BenoitSafari) [inline app/components/Forms/ConnectionForm.tsx:48]: If you name one *handler* with *handle* prefix you should do that with all handlers.
`submitSuccess` => `handleSubmitSuccess`

File: app/components/Forms/ConnectionForm.tsx [hunk 12]
```diff
@@ -0,0 +1,88 @@
+import React, { useRef } from 'react';
```
Comment (BenoitSafari) [inline app/components/Forms/ConnectionForm.tsx:1]:
```suggestion
import { useRef } from 'react';
```

File: app/components/FilmPageComponents/PresentationSection.tsx [hunk 13]
```diff
@@ -0,0 +1,185 @@
+import React from 'react';
```
Comment (BenoitSafari) [inline app/components/FilmPageComponents/PresentationSection.tsx:1]:
```suggestion
```

File: app/components/FilmPageComponents/CommentsSection.tsx [hunk 14]
```diff
@@ -0,0 +1,167 @@
+import React, { useState } from 'react';
```
Comment (BenoitSafari) [inline app/components/FilmPageComponents/CommentsSection.tsx:1]:
```suggestion
import { useState } from 'react';
```

**Review by BenoitSafari — CHANGES_REQUESTED** [review]

File: docker-compose.yml [hunk 15]
```diff
@@ -30,18 +35,33 @@ services:
       context: ./api
       dockerfile: Dockerfile.dev
     ports:
-      - 8080:8080 # API
+      - 3005:3005 # API
       - 5555:5555 # Prisma Studio
```
[the remark this answers is not in the document]
Comment (Julien-Goletto) [reply docker-compose.yml:39]: Deleted unnecessary lines at the same time.

File: app/pages/films/index.tsx [hunk 16]
```diff
@@ -0,0 +1,9 @@
+import React, { useState, useEffect } from 'react';
+
+export default function Films() {
```
[the remark this answers is not in the document]
Comment (Julien-Goletto) [reply app/pages/films/index.tsx:3]: Issue #66

File: app/pages/index.tsx [hunk 17]
```diff
@@ -1,13 +1,232 @@
...
+      <section
+        id="hero"
+        className={sectionStyle}>
+        <div className={sectionContentStyle + 'md:flex-col'}>
+          <div className="w-full flex justify-between">
+            <div className="flex flex-col gap-8 flex-1">
+              <h1 className="hero-text ">
+                Bienvenue dans votre <span className="text-orange-primary">ciné-club</span> virtuel !
+              </h1>
+              <h2 className={h2Style}>Chaque semaine, un film à découvrir</h2>
```
[the remark this answers is not in the document]
Comment (Julien-Goletto) [reply app/pages/index.tsx:55]: APP - Lighten the HomePage #68

File: app/pages/films/[film].tsx [hunk 18]
```diff
@@ -0,0 +1,294 @@
...
+  return (
+    <>
+      <Head>
+        <title>{`Cinévoraces - ${french_title}`}</title>
+        <meta
+          name="description"
+          content={`Découvrez le film ${french_title}`}
+        />
+      </Head>
+      <main className="custom-container ">
```
[the remark this answers is not in the document]
Comment (Julien-Goletto) [reply app/pages/films/[film].tsx:181]: Extra spaces to lighten classNames adding

File: app/components/FilmPageComponents/CommentsSection.tsx (hunk 14 above)
Comment (Julien-Goletto) [reply app/components/FilmPageComponents/CommentsSection.tsx:1]: Remove React import

File: app/components/Layout/HeaderMenu.tsx [hunk 19]
```diff
@@ -0,0 +1,118 @@
...
+            src={avatar_url ? avatar_url : defaultUserPic}
+            width={35}
+            height={35}
+            alt="User menu button"
+            className="user w-[35px] h-[35px] p-0.5 border border-white rounded-full lg:w-[45px] lg:h-[45px]"
+          />)
+      }
+    </button>
+  );
+};
```
[the remark this answers is not in the document]
Comment (Julien-Goletto) [reply app/components/Layout/HeaderMenu.tsx:50]: Dedicated Issue :  APP - More Component Decompositing #70

File: app/components/Input/Text.tsx [hunk 20]
```diff
@@ -0,0 +1,151 @@
...
+      <label htmlFor={id}>{label}</label>
+      <input
+        {...inputProps}
+        name={id}
+        className={basicStyles}
+      />
+      <p className="hidden peer-invalid:block text-sm font-light text-red-500">{errorMessage}</p>
+    </div>
+  );
+};
```
[the remark this answers is not in the document]
Comment (Julien-Goletto) [reply app/components/Input/Text.tsx:57]: APP - More Component Decompositing #70

File: app/components/Forms/ConnectionForm.tsx (hunk 11 above)
Comment (Julien-Goletto) [reply app/components/Forms/ConnectionForm.tsx:48]: Issue #71

**Review by BenoitSafari — APPROVED** [review]

**Pull request merged**